src/main/java/com/commercetools/sync/products/helpers/PriceReferenceResolver.java

```diff
@@ -56,18 +55,14 @@ public PriceReferenceResolver(@Nonnull final ProductSyncOptions options,
     }
 
     @Override
+    @Nonnull
     protected CompletionStage<PriceDraftBuilder> resolveCustomTypeReference(
-            @Nonnull final PriceDraftBuilder draftBuilder) {
-        final CustomFieldsDraft custom = draftBuilder.getCustom();
-        if (custom != null) {
-            return getCustomTypeId(custom,
-                    format(FAILED_TO_RESOLVE_CUSTOM_TYPE, draftBuilder.getCountry(), draftBuilder.getValue()))
-                .thenApply(resolvedTypeIdOptional -> resolvedTypeIdOptional
-                    .map(resolvedTypeId -> draftBuilder
-                        .custom(CustomFieldsDraft.ofTypeIdAndJson(resolvedTypeId, custom.getFields())))
-                    .orElse(draftBuilder));
-        }
-        return completedFuture(draftBuilder);
+        @Nonnull final PriceDraftBuilder draftBuilder) {
+
+        return resolveCustomTypeReference(draftBuilder,
+            PriceDraftBuilder::getCustom,
+            PriceDraftBuilder::custom,
+            format(FAILED_TO_RESOLVE_CUSTOM_TYPE, draftBuilder.getCountry(), draftBuilder.getValue()));
     }
 
```

Comment:
isn`t is enough to only pass the draftbuilder to that function, and extract the functions "getCuston" and "custom" from the given draftbuilder (anyhow)?

Reply:
no because all the builders do not implement an interface which has these `getCusom` and `custom` methods in the JVM SDK 😞 . Every builder has it's own inividual method (just with same names). That's why I have to pass the method references.